AI Meeting Recording Integration

jhodapp opened the pull request:
## Summary
Complete AI-powered meeting recording integration that enables automatic transcription and action item extraction from coaching sessions.

### Key Features
- **Recall.ai Integration**: Bot management for Google Meet recording
- **AssemblyAI Integration**: Audio transcription with speaker diarization
- **LeMUR Integration**: AI-powered action item extraction from transcripts
- **Dual Privacy Levels**: Independent coach and coachee AI privacy controls
- **AI Suggestions System**: Review and accept/dismiss extracted action items
- **Manual Extraction**: Trigger LeMUR extraction on-demand via API

### Technical Changes
- New database entities: `meeting_recordings`, `transcription_segments`, `ai_suggested_items`, `user_integrations`
- New migrations for AI privacy levels and LeMUR fields
- Webhook handlers for Recall.ai and AssemblyAI callbacks
- External API gateway clients with proper error handling
- Refactored `CreateAiSuggestion` struct to comply with clippy's parameter limits

## Test Plan
- [ ] Verify Recall.ai bot creation and webhook callbacks
- [ ] Verify AssemblyAI transcription and webhook handling
- [ ] Test LeMUR action extraction from completed transcripts
- [ ] Confirm AI suggestions appear and can be accepted/dismissed
- [ ] Test dual privacy level controls for coach and coachee
- [ ] Verify manual extraction endpoints work correctly

jhodapp commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
```diff
@@ -0,0 +1,919 @@
+    ├── ZoomProvider
+    └── TeamsProvider
+
+RecordingBotProvider        (Meeting bots)
```
What do you think of calling this `MeetingBotProvider` instead which may allow for these bots to be responsible for other kinds of things?

jhodapp commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
````diff
@@ -0,0 +1,919 @@
+```rust
+/// Configuration for OAuth authentication
+#[derive(Debug, Clone)]
+pub struct OAuthConfig {
````
Perhaps we should extract OAuth token handling into a separate crate that you can use here and I can use for my meeting link/creation manager crate that I'm working on. What do you think - worth designing to handle my use cases and yours and design this together?

Perhaps this becomes a generalized AuthenticationProvider trait that can have different implementations like GoogleOAuth, ZoomOAuth, Assembly, Recall, etc.

jhodapp commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
```diff
@@ -0,0 +1,919 @@
+pub struct ExtractedAgreement {
+    pub content: String,
+    pub source_text: String,
+    pub stated_by: String,
```
I'm wondering if we also update the Agreement model to include a "created_by" field that gets correlated to this field? It could become useful data to have and it'd also be easy to populate it today as well with whomever creates an Agreement through the manual UI method as well.

jhodapp commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
```diff
@@ -0,0 +1,919 @@
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TopicMention {
```
The purpose of this is to be able to keep track of all mentions of interesting topics? How does something become a mentioned topic worth extracting?

Where will this get used?

jhodapp commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
```diff
@@ -0,0 +1,919 @@
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TokenUsage {
+    pub input_tokens: u32,
+    pub output_tokens: u32,
```
u32 should be more than enough at about 4.2 billion, but is it worth making the u64 for future-proofing just in case? Or am I overthinking this?

jhodapp commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
````diff
@@ -0,0 +1,919 @@
+}
+```
+
+### 7. Workflow Orchestrator (Optional)
````
Is this meant to be like a state machine of the entire process? Why's it marked as optional?

calebbourg commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
```diff
@@ -0,0 +1,919 @@
+    ├── OpenAiProvider
+    └── ClaudeProvider
+
+WebhookHandler             (Event processing)
```
I didn't consider that yet. This, as of right now, is specific to the AI recording abstraction

calebbourg commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
```diff
@@ -0,0 +1,919 @@
+
+/// Information about a recording bot
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BotInfo {
```
It will be up to the implementation of the `RecordingBotProvide` to decide this

calebbourg commented on docs/implementation-plans/meeting-and-auth-abstraction-layers.md:
```diff
@@ -0,0 +1,1306 @@
+/// Trait for storing provider credentials (API keys)
+#[async_trait]
+pub trait CredentialStorage: Send + Sync {
+    async fn store(&self, user_id: &str, provider_id: &str, credentials: CredentialData) -> Result<(), StorageError>;
```
Would it be correct to assume that we would back this via a seaORM entity and store these in postgres?

calebbourg commented on docs/implementation-plans/meeting-and-auth-abstraction-layers.md:
```diff
@@ -0,0 +1,1306 @@
+use meeting_auth::TokenManager;
+
+/// Meeting space returned after creation
+pub struct MeetingSpace {
```
Similar to above, I think this should be just called `Space` and be imported using `as`

```rust
use meeting_manager::meeting::Space as MeetingSpace
```

jhodapp commented:
> @jhodapp I took a first pass at this. Overall I think it looks really great, very nice job!!
> 
> The main changes I would request are around naming and organization of the code. Apart from that, I don't have any major thoughts at this point. I think it's looking really good and will provide an excellent base to start with and we can make tweaks as we execute the actual implementation.
> 
> I have not had a chance to really delve into the database related things. I'd like to do that with a fresh pair of eyes.

Thanks for the great review @calebbourg. I addressed all of your feedback, and I also updated and made the new error types and handling part of the plan more explicit in commit 8a5382a.

calebbourg commented on docs/implementation-plans/meeting-ai-abstraction-layer.md:
```diff
@@ -0,0 +1,1274 @@
+    ├── DeepgramProvider
+    └── WhisperProvider
+
+AiAnalysisProvider           (Action extraction, summaries)
```
Great suggestion dcaf354658488236b820c850296e09a946320a4a
